### PR TITLE
Adds NewFromFloatWithRound to avoid precision loss

### PR DESCRIPTION
**Pull request by rfloriano**
### Problem

The existing `NewFromFloat` function has precision loss issues due to its behavior of always truncating trailing decimals down (using `int64()` conversion). This leads to unexpected results when converting precise decimal values:

```go
m := NewFromFloat(147.23, USD)
fmt.Println(m.Display()) // --> $147.22 (lost penny) ❌

m := NewFromFloat(-0.125, EUR) 
// Results in amount: -12 (truncated down)
```

This behavior is particularly problematic when:
- Working with exact decimal amounts that match the currency's precision (e.g. 147.23 - two decimal places with currency USD, Fraction = 2)
- Converting financial data where precision is critical
- Dealing with negative values where truncation becomes more counterintuitive

### Solution

This PR introduces `NewFromFloatWithRound`, a new function that uses proper mathematical rounding (`math.Round`) instead of truncation:

```go
// NewFromFloatWithRound creates and returns new instance of Money from a float64.
// It rounds trailing decimals to the nearest integer as math.Round does.
func NewFromFloatWithRound(amount float64, code string) *Money {
    currency := newCurrency(code).get()
    currencyDecimals := math.Pow10(currency.Fraction)
    return &Money{
        amount:   int64(math.Round(amount * currencyDecimals)),
        currency: currency,
    }
}
```

### Key Improvements

1. **Precision preservation**: Exact decimal values are preserved correctly
   ```go
   m := NewFromFloatWithRound(147.23, USD)
   fmt.Println(m.Display()) // --> $147.23 ✅
   ```

2. **Proper rounding behavior**: Uses banker's rounding (round half to even) via `math.Round`
   ```go
   NewFromFloatWithRound(-147.23, EUR)  // amount: -14723 (rounded)
   vs
   NewFromFloat(-147.23, EUR)           // amount: -14722 (truncated)
   ```

3. **Consistent with mathematical expectations**: Follows standard rounding rules instead of always rounding down

### Backward Compatibility

- The existing `NewFromFloat` function remains unchanged to maintain backward compatibility
- Applications that specifically need truncation behavior can continue using the original function
- The new function provides an opt-in solution for applications requiring proper rounding

### Usage Examples

```go
// Old behavior (truncation)
m1 := NewFromFloat(147.23, USD)      // $147.22
m2 := NewFromFloat(-0.125, EUR)      // -€0.12

// New behavior (proper rounding) 
m3 := NewFromFloatWithRound(147.23, USD)  // $147.23
m4 := NewFromFloatWithRound(-0.125, EUR)  // -€0.13
```

This enhancement addresses precision issues while maintaining full backward compatibility, giving developers the choice between truncation and proper rounding based on their specific use cases.

## Summary by Sourcery

Add NewFromFloatWithRound to handle float-to-money conversion using math.Round instead of truncation, maintaining backward compatibility, and update tests and documentation accordingly

New Features:
- Add NewFromFloatWithRound function for float-to-money conversion with proper mathematical rounding

Enhancements:
- Preserve existing NewFromFloat behavior for backward compatibility

Documentation:
- Update README to document the new rounding-based initialization method

Tests:
- Add unit tests for NewFromFloatWithRound covering positive, negative, and unregistered currency cases

### CODE REVIEW

File: money.go [hunk 1]
```diff
@@ -94,6 +94,17 @@ func NewFromFloat(amount float64, code string) *Money {
...
+// NewFromFloatWithRound creates and returns new instance of Money from a float64.
+// It rounds trailing decimals to the nearest integer as math.Round does.
+func NewFromFloatWithRound(amount float64, code string) *Money {
+	currency := newCurrency(code).get()
+	currencyDecimals := math.Pow10(currency.Fraction)
+	return &Money{
+		amount:   int64(math.Round(amount * currencyDecimals)),
+		currency: currency,
+	}
+}
```
[the remark this answers is not in the document]
Comment (rfloriano) [reply money.go:106]: It doesn't make sense. if we do it here, we probably should return an error, and in theory, we should also update the `NewFromFloat`

File: money_test.go [hunk 2]
```diff
@@ -841,6 +841,60 @@ func TestNewFromFloat_WithUnregisteredCurrency(t *testing.T) {
 	}
 }
```
[the remark this answers is not in the document]
Comment (rfloriano) [reply money_test.go:848]: done: https://github.com/Rhymond/go-money/pull/159/commits/73a978824ae9bc667db922fc601af857440c87d1

**Comment by rfloriano** [comment]
any updates here?
